File: cpp/250813_S4_10211.cpp
/**
 * https://www.acmicpc.net/problem/10211
 */
#include <ios>
#include <iostream>
#include <vector>
#include <limits>

using namespace std;

int msp(int N, vector<int> nums) {
	int maxValue = numeric_limits<int>::min();
	int curValue = 0;
	for (int i = 0; i < nums.size(); ++i) {
		curValue = max(nums[i], curValue + nums[i]);
		maxValue = max(maxValue, curValue);
	}
	return maxValue;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	
	int T;
	cin >> T;
	vector<int> results;
	while(T--) {
		int N;
		cin >> N;
		vector<int> nums(N);
		for (int i = 0; i < N; ++i) {
			cin >> nums[i];
		}
		results.push_back(msp(N, nums));
	}
	
	for(int result: results) {
		cout << result << endl;
	}
	return 0;
}

File: python/250813_S4_10211.py
# https://www.acmicpc.net/problem/10211
import sys

def get_input():
    return sys.stdin.readline().strip()

def kadain(N, nums):
    maxValue = -sys.maxsize - 1
    curValue = 0
    for num in nums:
        curValue = max(num, num + curValue)
        maxValue = max(maxValue, curValue)
    return maxValue;

if __name__ == "__main__":
    results = []
    T = int(get_input())
    for _ in range(T):
        N = int(get_input())
        nums = list(map(int, get_input().split()))
        results.append(kadain(N, nums))
    for result in results:
        print(result)
